vm/src/obj/objtype.rs

```diff
@@ -49,33 +115,19 @@ pub fn init(ctx: &PyContext) {
     extend_class!(&ctx, &ctx.type_type, {
         "__call__" => ctx.new_rustfunc(type_call),
         "__new__" => ctx.new_rustfunc(type_new),
-        "__mro__" => ctx.new_property(type_mro),
-        "__repr__" => ctx.new_rustfunc(type_repr),
-        "__prepare__" => ctx.new_rustfunc(type_prepare),
+        "__mro__" => ctx.new_property(PyClassRef::mro),
+        "__repr__" => ctx.new_rustfunc(PyClassRef::repr),
+        "__prepare__" => ctx.new_rustfunc(PyClassRef::prepare),
         "__getattribute__" => ctx.new_rustfunc(type_getattribute),
-        "__instancecheck__" => ctx.new_rustfunc(type_instance_check),
-        "__subclasscheck__" => ctx.new_rustfunc(type_subclass_check),
+        "__instancecheck__" => ctx.new_rustfunc(PyClassRef::instance_check),
+        "__subclasscheck__" => ctx.new_rustfunc(PyClassRef::subclass_check),
         "__doc__" => ctx.new_str(type_doc.to_string()),
-        "__dir__" => ctx.new_rustfunc(type_dir),
+        "__dir__" => ctx.new_rustfunc(PyClassRef::dir),
     });
 }
 
-fn type_mro(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
-    arg_check!(vm, args, required = [(cls, Some(vm.ctx.type_type()))]);
-    match _mro(cls.clone()) {
-        Some(mro) => Ok(vm.context().new_tuple(mro)),
-        None => Err(vm.new_type_error("Only classes have an MRO.".to_string())),
-    }
-}
-
-fn _mro(cls: PyObjectRef) -> Option<Vec<PyObjectRef>> {
-    if let Some(PyClass { ref mro, .. }) = cls.payload::<PyClass>() {
-        let mut mro = mro.clone();
-        mro.insert(0, cls.clone());
-        Some(mro)
-    } else {
-        None
-    }
+fn _mro(cls: &PyClassRef) -> Vec<PyObjectRef> {
+    cls.iter_mro().map(|x| x.clone()).collect()
```

Comment:
`.cloned()`?

Reply:
Yes, indeed, you can use here `cls.iter_mro().cloned()`. The clippy tool will hint at this.

Reply:
Aha! Thanks. It hadn't fully sunk in what ```cloned``` is yet.